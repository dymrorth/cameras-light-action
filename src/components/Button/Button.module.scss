@import 'styles/common';

$button-radius: 50px;

.button {
    position: relative;
    border-radius: $button-radius;
    background: var(--gradient-primary);
    color: var(--color-white);
    padding: $space-xs $space-m;
    box-shadow: 0;
    isolation: isolate;

    font-weight: bold;
    text-transform: uppercase;
    letter-spacing: 1.12px;
    user-select: none;

    border: none;
    cursor: pointer;
    
    display: inline-flex;
    align-items: center;
    justify-content: center;

    transition: transform var(--transition-duration) ease;

    &::after {
        content: '';
        position: absolute;
        inset: 0;
        background: inherit;
        z-index: -1;
        border-radius: $button-radius;
        filter: blur(1em);
        opacity: 0.3;
        transition: opacity var(--transition-duration) ease-in;
    }

    &:hover,
    &:focus {
        transform: scale(1.03);
        &::after {
            opacity: 0.7;
        }
    }

    &:active {
        transform: scale(0.97);
    }

    &:disabled {
        pointer-events: none;
        background: var(--color-grey-2);
        color: var(--color-grey-4);
    }
}
