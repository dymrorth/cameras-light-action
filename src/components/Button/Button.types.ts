export interface ButtonProps {
    isDisabled?: boolean
    onClick?: () => void
}