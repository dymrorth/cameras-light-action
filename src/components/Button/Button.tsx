import React from 'react'

import { ButtonProps } from './Button.types'
import styles from './Button.module.scss'

const Button: React.FC<ButtonProps> = ({ isDisabled = false, ...props }) => {
    return <button className={styles.button} {...props} disabled={isDisabled}></button>
}

export default Button
