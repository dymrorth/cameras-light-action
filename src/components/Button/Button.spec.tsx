import Button from './Button'

import { render, RenderResult, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

describe('<Button />', () => {
    describe('when mounted', () => {
        it('has inner text', () => {
            const innerText = 'Click me!'
            const { asFragment }: RenderResult = render(<Button>{innerText}</Button>)
            const button: HTMLElement = screen.getByText(innerText)

            expect(button).toBeDefined()
            expect(asFragment()).toMatchSnapshot()
        })
    })

    describe('when clicked', () => {
        it('calls handler function', () => {
            const handleClick = jest.fn()
            const innerText = 'Click me!'
            render(<Button onClick={handleClick}>{innerText}</Button>)
            const button: HTMLElement = screen.getByText(innerText)

            userEvent.click(button)
            expect(handleClick).toHaveBeenCalledTimes(1)
        })
    })
})
