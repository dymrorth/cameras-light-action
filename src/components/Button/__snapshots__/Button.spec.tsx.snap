// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> when mounted has inner text 1`] = `
<DocumentFragment>
  <button
    class="button"
  >
    Click me!
  </button>
</DocumentFragment>
`;
